overlord/boot/firstboot.go

```diff
@@ -126,6 +136,78 @@ func populateStateFromSeed() error {
 	return ovld.Stop()
 }
 
+func importAssertionsFromSeed(st *state.State) error {
+	st.Lock()
+	defer st.Unlock()
+
+	assertSeedDir := filepath.Join(dirs.SnapSeedDir, "assertions")
+	dc, err := ioutil.ReadDir(assertSeedDir)
+	if err != nil {
+		return fmt.Errorf("cannot read assert seed dir: %s", err)
+	}
+
+	// FIXME: remove this check once asserts are mandatory
+	if len(dc) == 0 {
+		return nil
+	}
+
+	// collect
+	var modelAssertion *asserts.Model
+	bs := asserts.NewMemoryBackstore()
+	db, err := asserts.OpenDatabase(&asserts.DatabaseConfig{
+		KeypairManager: asserts.NewMemoryKeypairManager(),
+		Backstore:      bs,
+		Trusted:        sysdb.Trusted(),
+	})
+	for _, fi := range dc {
+		content, err := ioutil.ReadFile(filepath.Join(assertSeedDir, fi.Name()))
+		if err != nil {
+			return fmt.Errorf("cannot read assertion: %s", err)
+		}
+		as, err := asserts.Decode(content)
+		if err != nil {
+			return fmt.Errorf("cannot decode assertion: %s", err)
+		}
+		if err := bs.Put(as.Type(), as); err != nil {
+			return err
+		}
+		if as.Type() == asserts.ModelType {
+			if modelAssertion != nil {
+				return fmt.Errorf("cannot add more than one model assertion")
+			}
+			modelAssertion = as.(*asserts.Model)
+		}
+	}
+	// verify we have one model assertion
+	if modelAssertion == nil {
+		return fmt.Errorf("need a model assertion")
+	}
+
+	// add all assertions to the database
+	retrieve := func(ref *asserts.Ref) (asserts.Assertion, error) {
+		as, err := ref.Resolve(db.Find)
+		if err != nil {
+			return nil, fmt.Errorf("cannot find %s: %s", ref.Unique(), err)
+		}
+		return as, nil
+	}
+	save := func(as asserts.Assertion) error {
+		return assertstate.Add(st, as)
+	}
+	fetcher := asserts.NewFetcher(db, retrieve, save)
+	if err := fetcher.Fetch(modelAssertion.Ref()); err != nil {
+		return err
+	}
+
+	// set device,model from the model assertion
+	auth.SetDevice(st, &auth.DeviceState{
+		Brand: modelAssertion.BrandID(),
+		Model: modelAssertion.Model(),
+	})
```

Comment:
this looks good, we need to think a bit where to keep things disabled until the server works (we have code to cope with no model, I'm thinking we have model assertions but the server is not quite there yet case). Atm devicestate is not plugged at all but not sure we can keep it that way